pkg/codegen/nodejs/gen.go

```diff
@@ -69,6 +76,30 @@ func camel(s string) string {
 	return string(res)
 }
 
+// Turns an identifier into PascalCase.
+func pascal(s string) string {
+	split := [][]rune{{}}
+	runes := []rune(s)
+	for _, r := range runes {
+		if !isLegalIdentifierPart(r) && r != '.' {
+			split = append(split, []rune{})
+		} else {
+			split[len(split)-1] = append(split[len(split)-1], r)
+		}
+	}
+	words := make([]string, len(split))
+	for i, v := range split {
+		words[i] = title(string(v))
+	}
+	return strings.Join(words, "")
+}
```

Comment:
Since there's already a `camel` function couldn't `pascal` just be `title(camel(s))`?

Reply:
No. I had that thought, but it doesn't work out. `camel` does conversions like `JSONObject` to `jsonObject`. `title(camel("JSONObject"))` would be `JsonObject` and we want it to remain unchanged. `camel` also has no impact on `-`, which we want `pascal` to do.

Reply:
Expanding on the comment for `pascal` to include what you just said would be awesome!